### PR TITLE
Run2 2017

### PR DESCRIPTION
**Pull request by rpatelCERN**
Sorry about that i made the PR to the wrong branch. I attach the same comments are as the previous PR:

Hey Kevin,
Here are a few changes that include:
Updating the scenarios.py for the _cff files for new 2017 collision datasets

Also I updated a few collection names based on the edm Collections in the latest data files.

I also updated the list of trigger names.
Let me know if you have any questions or suggestions

### CODE REVIEW

File: Production/test/runMakeTreeFromMiniAOD_cfg.py [hunk 1]
```diff
@@ -48,7 +47,7 @@
 
 #temporary redirector fix
 #fastsim signal is phedexed to LPC Tier3
-redir=parameters.value("redir", "root://cmseos.fnal.gov/" if fastsim and signal else "root://cmsxrootd.fnal.gov/")
+redir=parameters.value("redir", "root://eoscms.cern.ch/" if fastsim and signal else "root://eoscms.cern.ch/")
```
Comment (kpedro88) [inline Production/test/runMakeTreeFromMiniAOD_cfg.py]: nope, revert this
(the whole point of a command-line option is that you don't have to edit the file...)

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py [hunk 2]
```diff
@@ -447,15 +448,13 @@ def makeTreeFromMiniAOD(
             filterName  = cms.string("Flag_EcalDeadCellTriggerPrimitiveFilter"),
         )
         VarsInt.extend(['EcalDeadCellTriggerPrimitiveFilter'])
-        
         process.eeBadScFilter = filterDecisionProducer.clone(
-            trigTagArg1  = cms.string('TriggerResults'),
+            trigTagArg1  = cms.string("TriggerResults"),
```
Comment (kpedro88) [inline TreeMaker/python/makeTreeFromMiniAOD_cff.py]: Is there a reason this undoes the change in filterdecisionproducer_cfi.py? (and also for all the other MET filters...)

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py [hunk 3]
```diff
@@ -204,6 +204,7 @@ def makeTreeFromMiniAOD(
     # default miniAOD tags
     JetTag = cms.InputTag('slimmedJets')
     JetAK8Tag = cms.InputTag('slimmedJetsAK8')
+    #JetAK8Tag = cms.InputTag('slimmedJetsAK8PFPuppiSoftDropPacked')
```
Comment (kpedro88) [inline TreeMaker/python/makeTreeFromMiniAOD_cff.py]: delete commented-out code

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py [hunk 4]
```diff
@@ -708,36 +707,39 @@ def makeTreeFromMiniAOD(
     
     # AK8 jet variables - separate instance of jet properties producer
     from TreeMaker.Utils.jetproperties_cfi import jetproperties
+    #JetAK8Tag = cms.InputTag('slimmedJetsAK8')
```
Comment (kpedro88) [inline TreeMaker/python/makeTreeFromMiniAOD_cff.py]: delete commented-out code

File: Production/python/Run2017A-PromptReco-v3/HTMHT_cff.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import FWCore.ParameterSet.Config as cms
```
Comment (kpedro88) [inline Production/python/Run2017A-PromptReco-v3/HTMHT_cff.py]: 2017A does not contain any certified data (and never will, afaik) - please remove

File: Production/python/scenarios.py [hunk 6]
```diff
@@ -60,6 +60,13 @@ def __init__(self,sname):
                           jecfile="data/jec/Summer16_23Sep2016AllV3_DATA",
                           residual=True,
                           era="Run2_25ns")
+        elif sname == "2017PromptCollisions":
```
Comment (kpedro88) [inline Production/python/scenarios.py]: * call this 2017BC (GT, etc. may change for later run periods)
* describe this scenario in README.md
* add an example to unitTest.py

File: Production/python/scenarios.py [hunk 7]
```diff
@@ -60,6 +60,13 @@ def __init__(self,sname):
                           jecfile="data/jec/Summer16_23Sep2016AllV3_DATA",
                           residual=True,
                           era="Run2_25ns")
+        elif sname == "2017PromptCollisions":
+            self.set_vars(globaltag="92X_dataRun2_Prompt_v8",
+                          tagname="RECO",
+                          jsonfile="data/Cert_294927-299042_13TeV_PromptReco_Collisions17_JSON.txt",
+                          jecfile="data/jec/Summer16_23Sep2016AllV3_DATA",
```
Comment (kpedro88) [inline Production/python/scenarios.py]: I doubt this is the right JEC for 2017 data... it actually doesn't look like there are 2017 JECs at this point, so probably best to leave this as `""` for now (which will avoid reapplying JECs at all, and just use the online)

File: Production/python/scenarios.py [hunk 8]
```diff
@@ -60,6 +60,13 @@ def __init__(self,sname):
                           jecfile="data/jec/Summer16_23Sep2016AllV3_DATA",
                           residual=True,
                           era="Run2_25ns")
+        elif sname == "2017PromptCollisions":
+            self.set_vars(globaltag="92X_dataRun2_Prompt_v8",
+                          tagname="RECO",
+                          jsonfile="data/Cert_294927-299042_13TeV_PromptReco_Collisions17_JSON.txt",
+                          jecfile="data/jec/Summer16_23Sep2016AllV3_DATA",
+                          residual=True,
+                          era="Run2_25ns")
```
Comment (kpedro88) [inline Production/python/scenarios.py]: This should be `Run2_2017`

File: Production/python/Run2017B-PromptReco-v2/HTMHT_cff.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
+import FWCore.ParameterSet.Config as cms
```
Comment (kpedro88) [inline Production/python/Run2017B-PromptReco-v2/HTMHT_cff.py:1]: you need Run2017B-PromptReco-v1 also

File: Production/python/Run2017C-PromptReco-v2/MET_cff.py [hunk 10]
```diff
@@ -0,0 +1,69 @@
+import FWCore.ParameterSet.Config as cms
```
Comment (kpedro88) [inline Production/python/Run2017C-PromptReco-v2/MET_cff.py:1]: you need Run2017C-PromptReco-v1 also

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py [hunk 11]
```diff
@@ -708,36 +707,39 @@ def makeTreeFromMiniAOD(
...
-    process.JetsPropertiesAK8.bDiscriminatorSubjet1 = cms.vstring('SoftDrop','pfCombinedInclusiveSecondaryVertexV2BJetTags')
-    process.JetsPropertiesAK8.bDiscriminatorSubjet2 = cms.vstring('SoftDrop','pfCombinedInclusiveSecondaryVertexV2BJetTags')
-    process.JetsPropertiesAK8.bDiscriminatorCSV = cms.vstring('pfBoostedDoubleSecondaryVertexAK8BJetTags')
+    #specify userfloat
+    #process.JetsPropertiesAK8.prunedMass = cms.vstring('prunedMass')
+    process.JetsPropertiesAK8.prunedMass = cms.vstring('ak8PFJetsCHSValueMap:ak8PFJetsCHSPrunedMass')
+    process.JetsPropertiesAK8.NsubjettinessTau1 = cms.vstring('ak8PFJetsCHSValueMap:NjettinessAK8CHSTau1')
+    process.JetsPropertiesAK8.NsubjettinessTau2 = cms.vstring('ak8PFJetsCHSValueMap:NjettinessAK8CHSTau2')
+    process.JetsPropertiesAK8.NsubjettinessTau3 = cms.vstring('ak8PFJetsCHSValueMap:NjettinessAK8CHSTau3')
+    #process.JetsPropertiesAK8.bDiscriminatorSubjet1 = cms.vstring('SoftDrop','pfCombinedInclusiveSecondaryVertexV2BJetTags')
```
Comment (kpedro88) [inline TreeMaker/python/makeTreeFromMiniAOD_cff.py:729]: I think you can restore these by using `SoftDropPuppi`

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py [hunk 12]
```diff
@@ -708,36 +707,39 @@ def makeTreeFromMiniAOD(
...
-    #specify userfloats
-    process.JetsPropertiesAK8.prunedMass = cms.vstring('ak8PFJetsCHSPrunedMass')
-    process.JetsPropertiesAK8.NsubjettinessTau1 = cms.vstring('NjettinessAK8:tau1')
-    process.JetsPropertiesAK8.NsubjettinessTau2 = cms.vstring('NjettinessAK8:tau2')
-    process.JetsPropertiesAK8.NsubjettinessTau3 = cms.vstring('NjettinessAK8:tau3')
-    process.JetsPropertiesAK8.bDiscriminatorSubjet1 = cms.vstring('SoftDrop','pfCombinedInclusiveSecondaryVertexV2BJetTags')
-    process.JetsPropertiesAK8.bDiscriminatorSubjet2 = cms.vstring('SoftDrop','pfCombinedInclusiveSecondaryVertexV2BJetTags')
-    process.JetsPropertiesAK8.bDiscriminatorCSV = cms.vstring('pfBoostedDoubleSecondaryVertexAK8BJetTags')
+    #specify userfloat
+    #process.JetsPropertiesAK8.prunedMass = cms.vstring('prunedMass')
```
Comment (kpedro88) [inline TreeMaker/python/makeTreeFromMiniAOD_cff.py:724]: Just removed pruned mass and replace with `PuppiSoftDropMass` (userfloat `ak8PFJetsPuppiSoftDropMass`)

File: Utils/python/filterdecisionproducer_cfi.py [hunk 13]
```diff
@@ -1,8 +1,8 @@
 import FWCore.ParameterSet.Config as cms
 
 filterDecisionProducer = cms.EDProducer('FilterDecisionProducer',
-trigTagArg1  = cms.string('TriggerResults'),
-trigTagArg2  = cms.string(''),
+trigTagArg1  = cms.string('slimmedPatTrigger'),
```
Comment (kpedro88) [inline Utils/python/filterdecisionproducer_cfi.py]: this change breaks 2016 compatibility, so please remove all 2016 scenarios and input file lists (including from README, unit tests)

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py [hunk 14]
```diff
@@ -708,36 +707,39 @@ def makeTreeFromMiniAOD(
...
+    process.JetsPropertiesAK8.prunedMass = cms.vstring('ak8PFJetsCHSValueMap:ak8PFJetsCHSPrunedMass')
+    process.JetsPropertiesAK8.NsubjettinessTau1 = cms.vstring('ak8PFJetsCHSValueMap:NjettinessAK8CHSTau1')
```
Comment (kpedro88) [inline TreeMaker/python/makeTreeFromMiniAOD_cff.py:726]: I don't think these are right. It looks like the AK8 jet collection is now Puppi by default with PFCHS info stored as userfloats (reverse of the previous setup). So I think you want to use `NjettinessAK8Puppi:tau1`, etc.

(I turned on the debug flag in `JetProperties` and got this output:
```
subjetCollectionNames: SoftDropPuppi 
btagDiscriminatorNames: pfJetBProbabilityBJetTags pfJetProbabilityBJetTags pfTrackCountingHighEffBJetTags pfSimpleSecondaryVertexHighEffBJetTags pfSimpleInclusiveSecondaryVertexHighEffBJetTags pfCombinedSecondaryVertexV2BJetTags pfCombinedInclusiveSecondaryVertexV2BJetTags softPFMuonBJetTags softPFElectronBJetTags pfCombinedMVAV2BJetTags pfCombinedCvsLJetTags pfCombinedCvsBJetTags pfDeepCSVJetTags:probb pfDeepCSVJetTags:probc pfDeepCSVJetTags:probudsg pfDeepCSVJetTags:probbb pfBoostedDoubleSecondaryVertexAK8BJetTags pfBoostedDoubleSecondaryVertexAK8BJetTags 
userFloatNames: NjettinessAK8Puppi:tau1 NjettinessAK8Puppi:tau2 NjettinessAK8Puppi:tau3 ak8PFJetsCHSValueMap:NjettinessAK8CHSTau1 ak8PFJetsCHSValueMap:NjettinessAK8CHSTau2 ak8PFJetsCHSValueMap:NjettinessAK8CHSTau3 ak8PFJetsCHSValueMap:ak8PFJetsCHSPrunedMass ak8PFJetsCHSValueMap:ak8PFJetsCHSSoftDropMass ak8PFJetsCHSValueMap:eta ak8PFJetsCHSValueMap:mass ak8PFJetsCHSValueMap:phi ak8PFJetsCHSValueMap:pt ak8PFJetsPuppiSoftDropMass 
userIntNames: 
```
)

File: Utils/src/TriggerProducer.cc [hunk 15]
```diff
@@ -141,10 +141,12 @@ TriggerProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 
   //Find the matching triggers
   std::string testTriggerName;
+  //std::cout<<"Total Trigger Indices "<<parsedTrigNamesVec.size()<<std::endl;
```
Comment (kpedro88) [inline Utils/src/TriggerProducer.cc]: remove commented-out code

File: Utils/src/TriggerProducer.cc [hunk 16]
```diff
@@ -141,10 +141,12 @@ TriggerProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
...
   for(unsigned int parsedIndex = 0; parsedIndex < parsedTrigNamesVec.size(); parsedIndex++){
     trigNamesVec->at(parsedIndex) = parsedTrigNamesVec[parsedIndex];
     for(unsigned int trigIndex = 0; trigIndex < trigNames.size(); trigIndex++){
       testTriggerName = trigNames.triggerName(trigIndex);
+    //std::cout<<"Trigger Name/ index "<<testTriggerName<<" "<<trigIndex<<std::endl;;
```
Comment (kpedro88) [inline Utils/src/TriggerProducer.cc]: remove commented-out code

File: TreeMaker/python/doZinvBkg.py [hunk 17]
```diff
@@ -63,7 +63,6 @@ def reclusterZinv(process, geninfo, residual, cleanedCandidates, suff, fastsim,
             "NumChadrons"          ,
         )
     )
-    
     process.JetsPropertiesAK8Clean.prunedMass = cms.vstring('ak8PFJetsCHSCleanPrunedMass')
```
Comment (kpedro88) [inline TreeMaker/python/doZinvBkg.py:66]: make the corresponding AK8 changes here (wrt my comments for the main collection, e.g. CHS -> Puppi, etc.)

File: TreeMaker/python/triggerNameList.py [hunk 18]
```diff
@@ -1,49 +1,47 @@
...
-	'HLT_Photon165_HE10_v',
+	'HLT_Photon165_R9Id90_HE10_IsoM_v',
 	'HLT_Photon175_v',
-	'HLT_PFHT300_PFMET100_v',
-	'HLT_PFHT300_PFMET110_v',
-	'HLT_PFHT800_v',
-	'HLT_PFHT900_v',
-	'HLT_DoubleMu8_Mass8_PFHT300_v',
-	'HLT_DoubleEle8_CaloIdM_TrackIdM_Mass8_PFHT300_v',
+	'HLT_PFHT500_PFMET100_PFMHT100_IDTight_v',
```
Comment (kpedro88) [inline TreeMaker/python/triggerNameList.py:26]: Is this really the lowest-threshold HT-MHT trigger now? Wow...

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py (hunk 11 above)
Comment (rpatelCERN) [reply TreeMaker/python/makeTreeFromMiniAOD_cff.py:729]: Hey Kevin, For the subjettiness and the AK8 jet mass we are currently using the pruned version of these variables (which is what was stored in the previous versions of Treemaker). These variables are still stored in the current datasets as user floats (which I got from your debug flag) : 

ak8PFJetsCHSValueMap:NjettinessAK8CHSTau1
 ak8PFJetsCHSValueMap:NjettinessAK8CHSTau2
ak8PFJetsCHSValueMap:NjettinessAK8CHSTau3
 ak8PFJetsCHSValueMap:ak8PFJetsCHSPrunedMass

I conferred with Andrew and the above variables are the ones we want to use.

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py (hunk 14 above)
Comment (rpatelCERN) [reply TreeMaker/python/makeTreeFromMiniAOD_cff.py:726]: Hey Kevin, For the subjettiness and the AK8 jet mass we are currently using the pruned version of these variables (which is what was stored in the previous versions of Treemaker). These variables are still stored in the current datasets as user floats (which I got from your debug flag) :

ak8PFJetsCHSValueMap:NjettinessAK8CHSTau1
ak8PFJetsCHSValueMap:NjettinessAK8CHSTau2
ak8PFJetsCHSValueMap:NjettinessAK8CHSTau3
ak8PFJetsCHSValueMap:ak8PFJetsCHSPrunedMass

I conferred with Andrew and the above variables are the ones we want to use.

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py (hunk 12 above)
Comment (rpatelCERN) [reply TreeMaker/python/makeTreeFromMiniAOD_cff.py:724]: Hey Kevin, For the subjettiness and the AK8 jet mass we are currently using the pruned version of these variables (which is what was stored in the previous versions of Treemaker). These variables are still stored in the current datasets as user floats (which I got from your debug flag) :

ak8PFJetsCHSValueMap:NjettinessAK8CHSTau1
ak8PFJetsCHSValueMap:NjettinessAK8CHSTau2
ak8PFJetsCHSValueMap:NjettinessAK8CHSTau3
ak8PFJetsCHSValueMap:ak8PFJetsCHSPrunedMass

I conferred with Andrew and the above variables are the ones we want to use.

File: Utils/python/filterdecisionproducer_cfi.py (hunk 13 above)
Comment (rpatelCERN) [reply Utils/python/filterdecisionproducer_cfi.py]: This change is reverted, I checked and the "TriggerResults" collection works

File: Production/test/runMakeTreeFromMiniAOD_cfg.py (hunk 1 above)
Comment (rpatelCERN) [reply Production/test/runMakeTreeFromMiniAOD_cfg.py]: reverted

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py (hunk 3 above)
Comment (rpatelCERN) [reply TreeMaker/python/makeTreeFromMiniAOD_cff.py]: removed

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py (hunk 2 above)
Comment (rpatelCERN) [reply TreeMaker/python/makeTreeFromMiniAOD_cff.py]: reverted

File: Production/python/scenarios.py (hunk 7 above)
Comment (rpatelCERN) [reply Production/python/scenarios.py]: removed the JEC file. I agree we can go with the online jet energy instead

File: Production/python/scenarios.py (hunk 8 above)
Comment (rpatelCERN) [reply Production/python/scenarios.py]: I added the new JSON: https://cms-service-dqm.web.cern.ch/cms-service-dqm/CAF/certification/Collisions17/13TeV/PromptReco/Cert_294927-299649_13TeV_PromptReco_Collisions17_JSON.txt

File: TreeMaker/python/doZinvBkg.py (hunk 17 above)
Comment (rpatelCERN) [reply TreeMaker/python/doZinvBkg.py:66]: We are still sticking with pruned Mass for the boosted analyses, so for the control regions we still use the Pruned Mass (cleaned in the above for the photons)

File: TreeMaker/python/triggerNameList.py (hunk 18 above)
Comment (rpatelCERN) [reply TreeMaker/python/triggerNameList.py:26]: We also have MET/MHT only triggers which can be used for the signal trigger for lower MET regions :

        'HLT_PFMETNoMu110_PFMHTNoMu110_IDTight_v',
        'HLT_PFMETNoMu120_PFMHTNoMu120_IDTight_v',
        'HLT_PFMETNoMu130_PFMHTNoMu130_IDTight_v',
        'HLT_PFMETNoMu140_PFMHTNoMu140_IDTight_v',
        'HLT_PFMET110_PFMHT110_IDTight_v',
        'HLT_PFMET120_PFMHT120_IDTight_v',
        'HLT_PFMET130_PFMHT130_IDTight_v',
        'HLT_PFMET140_PFMHT140_IDTight_v',

File: Production/python/Run2017A-PromptReco-v3/HTMHT_cff.py (hunk 5 above)
Comment (rpatelCERN) [reply Production/python/Run2017A-PromptReco-v3/HTMHT_cff.py]: Yes that is true, I removed that folder

File: Production/python/Run2017B-PromptReco-v2/HTMHT_cff.py (hunk 9 above)
Comment (rpatelCERN) [reply Production/python/Run2017B-PromptReco-v2/HTMHT_cff.py:1]: I was finding that the updated JSON does have runs in v2 of the 2017RunB dataset

File: Production/python/Run2017C-PromptReco-v2/MET_cff.py (hunk 10 above)
Comment (rpatelCERN) [reply Production/python/Run2017C-PromptReco-v2/MET_cff.py:1]: You are correct here, the current JSON will only have runs in v1.

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py (hunk 14 above)
Comment (kpedro88) [reply TreeMaker/python/makeTreeFromMiniAOD_cff.py:726]: These prompt 2017 ntuples won't be used for the boosted Higgs + MET analysis, will they? In the latest miniAOD, the default AK8 jets are now Puppi, so it seems like we should use the associated products from those jets. For 2016, we are still using PFCHS.

File: Production/python/Run2017B-PromptReco-v2/HTMHT_cff.py (hunk 9 above)
Comment (kpedro88) [reply Production/python/Run2017B-PromptReco-v2/HTMHT_cff.py:1]: There are certified runs in 2017B-v1 also, please add it

**Comment by rpatelCERN** [comment]
Hey Kevin, 
Actually we were also asked to check the behaviour of our mass variable (which is the pruned mass) for the latest data. So for now we would like to keep the Subjettiness and the mass values as below. 

-Rishi

**Comment by kpedro88** [comment]
okay, then we'll need to keep in mind to fix it to be congruent with the new default later

**Comment by rpatelCERN** [comment]
Okay sure I added a folder:
Run2017B-23Jun2017-v1
to Production/python

You should now see the commit in the PR. 

-Rishi

**Comment by kpedro88** [comment]
@rpatelCERN I don't think that's the right one - we should probably just use Run2017B-PromptReco-v1, unless you explicitly know what the 23Jun2017 one is

**Comment by rpatelCERN** [comment]
Okay I think there might have been some Trigger updates between those datasets but I am not sure. I added Run2017B-PromptReco-v1 and removed the 23Jun2017.

**Comment by kpedro88** [comment]
you didn't remove the 23Jun2017 yet...

**Comment by rpatelCERN** [comment]
Fixed!

File: Utils/python/prescaleweightproducer_cfi.py [hunk 19]
```diff
@@ -6,6 +6,6 @@
 
    prescales = cms.InputTag("patTrigger"),
 
-   objects = cms.InputTag("selectedPatTrigger"),
+   objects = cms.InputTag("slimmedPatTrigger"),
```
Comment (kpedro88) [inline Utils/python/prescaleweightproducer_cfi.py]: this change also breaks 2016 compatibility, so please revert it if possible or remove all 2016 scenarios and input file lists (including from README, unit tests)

**Comment by rpatelCERN** [comment]
That change is unfortunately necessary, so I commit a version that removes the 2016 datasets (MC/2016Data) and made the changes to the scenarios.py and the unitest.py

**Comment by kpedro88** [comment]
also update readme

**Comment by kpedro88** [comment]
in principle, stuff in Production/test and Production/test/condorSub should also be updated

**Comment by rpatelCERN** [comment]
I added a commit to address this

**Comment by rpatelCERN** [comment]
I added a commit to address this

File: Production/test/condorSub/dict_2017B.py [hunk 20]
```diff
@@ -0,0 +1,11 @@
+flist = {
+    "scenario": "2017PromptCollisions",
+    "samples": [
+        ['Run2017B-PromptReco-v2.HTMHT'],
```
Comment (kpedro88) [inline Production/test/condorSub/dict_2017B.py:4]: You should also be submitting 2017B-v1 (and eventually 2017C-v2, though none of it is certified yet)...

**Comment by kpedro88** [comment]
For the lumi calculation requested by @awhitbeck, you should also update https://github.com/TreeMaker/TreeMaker/blob/Run2_2017/Production/test/dataSamples.py

**Comment by rpatelCERN** [comment]
Maybe Troy can do that, since he got the lumiCalc script to work?

-Rishi

**Comment by rpatelCERN** [comment]
Sure I can do this.

-Troy

**Comment by kpedro88** [comment]
going to merge this now, last few comments should be addressed in a followup PR

**Comment by rpatelCERN** [comment]
Thanks!

**Pull request merged**